Libplanet/HashDigest.cs

```diff
@@ -77,6 +77,36 @@ public HashDigest(byte[] hashDigest)
             _byteArray = hashDigest.ToImmutableArray();
         }
 
+         /// <summary>
+         /// Converts a <see cref="ImmutableArray"/> of <see cref="byte"/> array into a
+         /// <see cref="HashDigest{T}"/>.
+         /// </summary>
+         /// <param name="hashDigest">A <see cref="byte"/> array that encodes
+         /// a <see cref="HashDigest{T}"/>.  It must not be <c>null</c>,
+         /// and its <see cref="Array.Length"/> must be the same to
+         /// <see cref="Size"/>.</param>
+         /// <exception cref="ArgumentNullException">Thrown when the given
+         /// <paramref name="hashDigest"/> is <c>null</c>.</exception>
+         /// <exception cref="ArgumentOutOfRangeException">Thrown when the given
+         /// <paramref name="hashDigest"/>'s <see cref="Array.Length"/> is not
+         /// the same to the <see cref="Size"/> the hash algorithm
+         /// (i.e., <typeparamref name="T"/> requires.</exception>
+        public HashDigest(ImmutableArray<byte> hashDigest)
```

Comment:
As we have this `HashDigest<ImmutableArray<byte>)` constructor now, the existing `HashDigest(byte[])` constructor could be reduced like:

```csharp
public HashDigest(byte[] hashDigest)
    : this((hashDigest ?? throw new ArgumentNullException(nameof(hashDigest))).ToImmutableArray())
{
}
```

Reply:
awesome! I'll change like that.